### PR TITLE
Add output format parameter to get_stats19 function and add format_ppp function. Fix #134

### LINKED ISSUE #134: Add output_format argument to get_stats19()
I can think of various pros, as it makes the data more usable more quickly without knowing about `format_sf()` but there maybe cons of doing this. To discuss tomorrow.

Heads-up @layik, @mem48, @wengraf, @publichealthgeek and @agila5, seem like a good plan to you?

### PR DESCRIPTION
**Pull request by agila5**
As the title says, I added a format_ppp function (to transform the STATS19 data into ppp format) and, after the discussion in #134 , I added an output_format parameter to get_stats19 function. The possible values are "tibble", "sf" or "ppp".

### CODE REVIEW

**Comment by Robinlovelace** [comment]
Just testing this with:

```
git checkout -b agila5-add_output_format master
git pull https://github.com/agila5/stats19.git add_output_format
```

**Comment by agila5** [comment]
Actually I forgot to run checks and there are several errors 😅  I will fix them asap.

**Comment by Robinlovelace** [comment]
OK will let travis catch the easy ones... I think 

```
usethis::use_package("spatstat", type = "Suggests")
```

May be needed.

**Comment by Robinlovelace** [comment]
Something tells me `spatstat` could benefit from a more efficient plotting method...

```
a = stats19::get_stats19(2018, output_format = "ppp")
Files identified: dftRoadSafetyData_Accidents_2018.csv

   http://data.dft.gov.uk.s3.amazonaws.com/road-accidents-safety-data/dftRoadSafetyData_Accidents_2018.csv
Attempt downloading from: 
trying URL 'http://data.dft.gov.uk.s3.amazonaws.com/road-accidents-safety-data/dftRoadSafetyData_Accidents_2018.csv'
Content type 'application/octet-stream' length 16827907 bytes (16.0 MB)
==================================================
downloaded 16.0 MB

Data saved at /tmp/RtmpQKxWE4/dftRoadSafetyData_Accidents_2018.csv
Reading in: 
/tmp/RtmpQKxWE4/dftRoadSafetyData_Accidents_2018.csv
55 rows removed with no coordinates
Warning message:
some mark values are NA in the point pattern x 
> plot(a)
Error: cannot allocate vector of size 112.0 Gb
```

**Comment by agila5** [comment]
> May be needed.

Should be fine now. I added also osmdata to Suggests because the last example (which is really good IMO) requires `osmdata::getbb`. Maybe we can just comment that example out and exclude osmdata dependency, not sure. 

> Something tells me spatstat could benefit from a more efficient plotting method...

Actually that occurrs only because the default options of plot.ppp display all the marks for all the points. I added `use.marks = FALSE` to all example to avoid that problem.

**Comment by layik** [comment]
Hi @agila5, hello @Robinlovelace I think you both know it is simple to implement #136. Apologies as I cannot contribute as much as might be needed today/these days. 

Please allow me to say that I see no reason for this over-complication, both the issue and also introducing "ppp". 

I genuinely do not want to understate the effort Andrea and thank you for the effort.

**Comment by agila5** [comment]
> Please allow me to say that I see no reason for this over-complication, both the issue and also introducing "ppp".

I should have asked before writing all that stuff, you're right 😅 I added that option because I see no reason to prefer sf output to all the other possibilities. Nevertheless I can just exclude that options if you think it's unnecessarily complicating stats19 code. @Robinlovelace what do you think? 

Not sure why the travis checks fail because I see no error when I run those check in locale, even if I add "--as-cran --run-donttest" to check options.

**Comment by Robinlovelace** [comment]
I think this change, which makes it easier for `spatstats` users, some of whom are leading experts in the statistical analysis of point pattern data, will encourage more people to use the valuable data provided by `stats19` package so it's a big :+1:  from me. Because the default is a data frame this will be fully backward compatible with older versions.

Question for @layik and @agila5: is it worth providing helper functions such as `get_stats19_vehicles()`, `get_stats19_casualties()` and `get_stats19_sf()` to increase the visibility of alternative datasets and output formats?

**Comment by Robinlovelace** [comment]
Regarding Travis, @agila5 I think it's the examples, e.g.: https://travis-ci.org/ropensci/stats19/builds/628813089#L2169

Overall great work, I'm confident this will increase the usefulness of the package for some people at least.

File: DESCRIPTION [hunk 1]
```diff
@@ -52,9 +52,11 @@ Suggests:
     htmltools,
     tmap,
     jsonlite,
-    pct
+    pct,
+    spatstat,
+    osmdata
```
Comment (Robinlovelace) [inline DESCRIPTION]: We should also add geofabric when it's on CRAN :pray:

File: tests/testthat/test-format.R [hunk 2]
```diff
@@ -54,3 +54,33 @@ test_that("format_sf works", {
   expect_true(is(df1, "sf"))
   expect_true(is(df2, "sf"))
 })
+
+test_that("format_ppp returns ppp object", {
```
Comment (Robinlovelace) [inline tests/testthat/test-format.R]: Great to see these tests in there, many thanks for this attention to detail @agila5

**Comment by agila5** [comment]
> Regarding Travis, @agila5 I think it's the examples, e.g.: https://travis-ci.org/ropensci/stats19/builds/628813089#L2169

It should be fixed now. The problem occurred with `osmdata::getbb` so I just commented out that example (and, in this case, we could also remove osmdata from Suggests). I don't like this solution but I couldn't think of anything easier.

> Question for @layik and @agila5: is it worth providing helper functions such as get_stats19_vehicles(), get_stats19_casualties() and get_stats19_sf() to increase the visibility of alternative datasets and output formats?

In my opinion no since it's better to provide additional parameters to `get_stats19()` function to customize its output instead of several functions. 

> I think this change, which makes it easier for spatstats users, some of whom are leading experts in the statistical analysis of point pattern data, will encourage more people to use the valuable data provided by stats19 package so it's a big 👍 from me. Because the default is a data frame this will be fully backward compatible with older versions.

I totally understand @layik point of view saying that I included a lot of unnecessary changes to stats19 code. Maybe we could wait a few weeks to merge this commit and ask (via twitter or similar tools) to the people using stats19 data if they are interested in these ppp interface and if they think the format_ppp function could be useful. 

Another problem (which is not a problem IMO since it's documented but I just need to make this point clear) is that if a user runs `get_stats19()` function setting `output_format` not equal to `tibble`, `sf` or `ppp` (i.e. the last test) then the output is NULL (because of the switch function). What do you think on this point?

**Comment by Robinlovelace** [comment]
> Maybe we could wait a few weeks to merge this commit and ask (via twitter or similar tools) to the people using stats19 data if they are interested in these ppp interface and if they think the format_ppp function could be useful.

Just merged it, I'm confident it's worth adding, as discussed and I think agreed on in https://github.com/ropensci/stats19/issues/134 - we can tidy-up + discuss before releasing this on CRAN is my thinking.

**Comment by Robinlovelace** [comment]
> Another problem (which is not a problem IMO since it's documented but I just need to make this point clear) is that if a user runs get_stats19() function setting output_format not equal to tibble, sf or ppp (i.e. the last test) then the output is NULL (because of the switch function). What do you think on this point?

I think that this is such an unusual edge case that it's not worth spending too much time on. However, I think a warning like `output_format not supported, using default tibble output format` would be slightly better than providing `NULL` as an output, I cannot see how that will useful. That would mean changing 1 line in the code and changing this:

```
acc = get_stats19(year = 2009, type = "acc", output_format = "abcdef")
  expect_null(acc)
```

to something like this


```
acc = get_stats19(year = 2009, type = "acc", output_format = "abcdef")
  expect_true(is(acc, "tibble"))
```

**Comment by agila5** [comment]
Ok, awesome, thanks.
